src/nvim/main.c

```diff
@@ -332,6 +332,12 @@ int main(int argc, char **argv)
   /* Source startup scripts. */
   source_startup_scripts(&params);
 
+  // If using the runtime (-u is not NONE), enable syntax and filetype plugins
+  if (params.use_vimrc != NULL && strcmp(params.use_vimrc, "NONE") != 0) {
+    syn_cmd("syntax");        // Enable syntax highlighting.
```

Comment:
`force_enable_filetype` should come first, because when it loads `syntax/syntax.vim` that will cause `filetype on` to run.